autogpt_platform/backend/backend/blocks/google/sheets.py

```diff
@@ -0,0 +1,192 @@
+            categories={BlockCategory.DATA},
+            input_schema=GoogleSheetsReadBlock.Input,
+            output_schema=GoogleSheetsReadBlock.Output,
+            disabled=not GOOGLE_OAUTH_IS_CONFIGURED,
+            test_input={
+                "spreadsheet_id": "1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms",
+                "range": "Sheet1!A1:B2",
+                "credentials": TEST_CREDENTIALS_INPUT,
+            },
+            test_credentials=TEST_CREDENTIALS,
+            test_output=[
+                (
+                    "result",
+                    [
+                        ["Name", "Score"],
+                        ["Alice", "85"],
+                    ],
+                ),
+            ],
+            test_mock={
+                "_read_sheet": lambda *args, **kwargs: [
+                    ["Name", "Score"],
+                    ["Alice", "85"],
+                ],
+            },
+        )
+
+    def run(
+        self, input_data: Input, *, credentials: GoogleCredentials, **kwargs
+    ) -> BlockOutput:
+        try:
+            service = self._build_service(credentials, **kwargs)
+            data = self._read_sheet(
+                service, input_data.spreadsheet_id, input_data.range
+            )
+            yield "result", data
+        except Exception as e:
+            yield "error", str(e)
+
+    @staticmethod
+    def _build_service(credentials: GoogleCredentials, **kwargs):
+        creds = Credentials(
+            token=(
+                credentials.access_token.get_secret_value()
+                if credentials.access_token
+                else None
+            ),
+            refresh_token=(
+                credentials.refresh_token.get_secret_value()
+                if credentials.refresh_token
+                else None
+            ),
+            token_uri="https://oauth2.googleapis.com/token",
+            client_id=kwargs.get("client_id"),
+            client_secret=kwargs.get("client_secret"),
+            scopes=credentials.scopes,
+        )
+        return build("sheets", "v4", credentials=creds)
+
+    def _read_sheet(self, service, spreadsheet_id: str, range: str) -> list[list[str]]:
+        sheet = service.spreadsheets()
+        result = sheet.values().get(spreadsheetId=spreadsheet_id, range=range).execute()
+        return result.get("values", [])
+
+
+class GoogleSheetsWriteBlock(Block):
+    class Input(BlockSchema):
+        credentials: GoogleCredentialsInput = GoogleCredentialsField(
+            ["https://www.googleapis.com/auth/spreadsheets"]
+        )
+        spreadsheet_id: str = SchemaField(
+            description="The ID of the spreadsheet to write to",
+        )
+        range: str = SchemaField(
+            description="The A1 notation of the range to write",
+        )
+        values: list[list[str]] = SchemaField(
+            description="The data to write to the spreadsheet",
+        )
```

Comment:
A `GoogleSheetsAppendRowBlock` would be an extremely useful addition next to this one.

Reply:
yes but not in this pr